src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -18,96 +18,132 @@
         {
-            ProjectInstance project = null;
-            var globalProperties = GetGlobalPropertiesFromUserArgs(parseResult);
+            // Setup
+            Debug.Assert(_propertyToCheck == MSBuildPropertyNames.PUBLISH_RELEASE || _propertyToCheck == MSBuildPropertyNames.PACK_RELEASE, "Only PackRelease or PublishRelease are currently expected.");
+            var nothing = Enumerable.Empty<string>();
+            if (String.Equals(Environment.GetEnvironmentVariable(EnvironmentVariableNames.DISABLE_PUBLISH_AND_PACK_RELEASE), "true", StringComparison.OrdinalIgnoreCase))
+            {
+                return nothing;
+            }
+
+            // Analyze Global Properties
+            var globalProperties = GetUserSpecifiedExplicitMSBuildProperties();
+            globalProperties = InjectTargetFrameworkIntoGlobalProperties(globalProperties);
 
-            if (parseResult.HasOption(configOption) || globalProperties.ContainsKey(MSBuildPropertyNames.CONFIGURATION))
-                yield break;
+            // Configuration doesn't work in a .proj file, but it does as a global property.
+            // Detect either A) --configuration option usage OR /p:Configuration=Foo, if so, don't use these properties.
+            if (_options.ConfigurationOption != null || globalProperties.ContainsKey(MSBuildPropertyNames.CONFIGURATION))
+                return new List<string> { $"-property:{EnvironmentVariableNames.DISABLE_PUBLISH_AND_PACK_RELEASE}=true" }; // Don't throw error if publish* conflicts but global config specified.
 
-            // CLI Configuration values take precedence over ones in the project. Passing PublishRelease as a global property allows it to take precedence.
-            project = GetTargetedProject(slnOrProjectArgs, globalProperties, defaultedConfigurationProperty);
+            // Determine the project being acted upon
+            ProjectInstance? project = GetTargetedProject(globalProperties);
 
+            // Determine the correct value to return
             if (project != null)
             {
-                string configurationToUse = "";
-                string releasePropertyFlag = project.GetPropertyValue(defaultedConfigurationProperty);
-                if (!string.IsNullOrEmpty(releasePropertyFlag))
-                    configurationToUse = releasePropertyFlag.Equals("true", StringComparison.OrdinalIgnoreCase) ? MSBuildPropertyNames.CONFIGURATION_RELEASE_VALUE : "";
+                string propertyToCheckValue = project.GetPropertyValue(_propertyToCheck);
+                if (!string.IsNullOrEmpty(propertyToCheckValue))
+                {
+                    var newConfigurationArgs = new List<string> {
+                        $"-property:{MSBuildPropertyNames.CONFIGURATION}={
+                            (propertyToCheckValue.Equals("true", StringComparison.OrdinalIgnoreCase) ? MSBuildPropertyNames.CONFIGURATION_RELEASE_VALUE : MSBuildPropertyNames.CONFIGURATION_DEBUG_VALUE)
```

Comment:
Looking at this, I think that we only want to explicitly set the Configuration to Release if the propertyToCheck is true.  If propertyToCheck is set to false, then we don't need to explicitly set the Configuration to Debug, we can just let the default value take effect, which is normally going to be Debug.

Does this make sense?  I'm not sure if I've considered this before.

Reply:
This is definitely something we could do. With that being said, I think this is preferable. If someone adds `Configuration` to the project file, it will impact the output path if this code for `Debug` is removed. Obviously, this is a bad operation in the first place, but it will not impact the output path with this change added.

Reply:
Ah... that's pretty bad. Changed, thanks for catching this